File: app/src/main/java/org/ngbp/jsonrpc4jtestharness/rpc/queryDeviceInf/IQueryDeviceInfo.java
package org.ngbp.jsonrpc4jtestharness.rpc.queryDeviceInf;

import com.github.nmuzhichin.jsonrpc.annotation.JsonRpcMethod;
import com.github.nmuzhichin.jsonrpc.annotation.JsonRpcType;

import org.ngbp.jsonrpc4jtestharness.rpc.queryDeviceInf.model.DeviceInfoParams;

@JsonRpcType
public interface IQueryDeviceInfo {
    @JsonRpcMethod("org.atsc.query.deviceInfo")
    DeviceInfoParams queryDeviceInfo();
}

File: app/src/main/java/org/ngbp/jsonrpc4jtestharness/rpc/queryDeviceInf/IQueryDeviceInfo.kt
package org.ngbp.jsonrpc4jtestharness.rpc.queryDeviceInf

import com.github.nmuzhichin.jsonrpc.annotation.JsonRpcMethod
import com.github.nmuzhichin.jsonrpc.annotation.JsonRpcParam
import com.github.nmuzhichin.jsonrpc.annotation.JsonRpcType
import org.ngbp.jsonrpc4jtestharness.rpc.queryDeviceInf.model.DeviceInfo
import org.ngbp.jsonrpc4jtestharness.rpc.queryDeviceInf.model.DeviceInfoParams

@JsonRpcType
interface IQueryDeviceInfo {
    @JsonRpcMethod("org.atsc.query.deviceInfo")
    fun queryDeviceInfo(@JsonRpcParam(value = "deviceInfoProperties", nullable = true) deviceInfoParams: List<String>?): DeviceInfo
}